src/main/java/br/com/sbk/sbking/networking/kryonet/KryonetServerListenerFactory.java

```diff
@@ -0,0 +1,116 @@
+package br.com.sbk.sbking.networking.kryonet;
+
+import static br.com.sbk.sbking.logging.SBKingLogger.LOGGER;
+
+import java.util.UUID;
+
+import com.esotericsoftware.kryonet.Connection;
+import com.esotericsoftware.kryonet.FrameworkMessage;
+import com.esotericsoftware.kryonet.Listener;
+import com.esotericsoftware.kryonet.Server;
+
+import br.com.sbk.sbking.networking.kryonet.messages.SBKingMessage;
+import br.com.sbk.sbking.networking.kryonet.messages.ServerToClient.TextMessage;
+
+public class KryonetServerListenerFactory {
+
+  public static Listener getServerListener(Server server) {
+    return new Listener() {
+      public void connected(Connection connection) {
+        LOGGER.debug("Entered --connected-- lifecycle method.");
+
+        KryonetSBKingServer kryonetSBKingServer = null;
+        ConnectionWithIdentifier connectionWithPlayer = null;
+        try {
+          kryonetSBKingServer = (KryonetSBKingServer) server;
+        } catch (Exception e) {
+          LOGGER.fatal("Client should be a KryonetSBKingServer");
+          LOGGER.fatal(server);
+          return;
+        }
+        try {
+          connectionWithPlayer = (ConnectionWithIdentifier) connection;
+        } catch (Exception e) {
+          LOGGER.fatal("connection should be a ConnectionWithPlayer");
+          LOGGER.fatal(connection);
+          return;
+        }
+
+        kryonetSBKingServer.addConnection(connectionWithPlayer);
+
+        TextMessage response = new TextMessage("You just connected! Welcome to SBKing!");
+        connection.sendTCP(response);
+      }
+
+      public void received(Connection connection, Object object) {
+        LOGGER.info("Entered --received-- lifecycle method.");
+
+        KryonetSBKingServer kryonetSBKingServer = null;
+        SBKingMessage message = null;
+        ConnectionWithIdentifier connectionWithPlayer = null;
+
+        if (object instanceof FrameworkMessage) {
```

Comment:
Are all FrameworkMessage pings?

And just as a curiosity, how frequent are those pings from kryonet?

Reply:
[This](https://github.com/EsotericSoftware/kryonet/blob/5665e794200878bbeda65e537f7a3348b11167f3/src/com/esotericsoftware/kryonet/FrameworkMessage.java) is the interface used. I only received `FrameworkMessage$KeepAlive` but didn't look out in the code if other messages are send. I guess we will discover in the logs when the first error happen :) 
The period seems to be defined in [this file](https://github.com/EsotericSoftware/kryonet/blob/5665e794200878bbeda65e537f7a3348b11167f3/src/com/esotericsoftware/kryonet/TcpConnection.java#L20) and it is 8s.